evm_loader/cli/src/main.rs

```diff
@@ -333,221 +336,372 @@ pub fn keccak256_h256(data: &[u8]) -> H256 {
         (address, seed)
     };
-
     debug!("Create code account: {}", &program_code.to_string());
 
-    let make_create_account_instruction = |acc: &Pubkey, ether: &H160, nonce: u8, balance: u64| {
-        Instruction::new(
+    // Check program account to see if partial initialization has occurred
+    if let Some(_account) = config.rpc_client.get_account_with_commitment(&program_id, CommitmentConfig::confirmed())?.value
+    {
+        return Err("Account already exist".to_string().into());
+        // debug!("Account already exist");
+    } else {
+        let mut instructions = Vec::new();
+        instructions.push(system_instruction::create_account_with_seed(
+            &creator.pubkey(), 
+            &program_code, 
+            &creator.pubkey(), 
+            &program_seed, 
+            minimum_balance_for_code, 
+            program_code_len as u64, 
+            &config.evm_loader)
+        );
+        instructions.push(Instruction::new_with_bincode(
             config.evm_loader,
-            &(2u32, balance, 0 as u64, ether.as_fixed_bytes(), nonce),
+            &(2u32, minimum_balance_for_account, 0u64, program_ether.as_fixed_bytes(), program_nonce),
             vec![AccountMeta::new(creator.pubkey(), true),
-                 AccountMeta::new(*acc, false),
+                 AccountMeta::new(program_id, false),
                  AccountMeta::new(program_code, false),
                  AccountMeta::new_readonly(system_program::id(), false),]
-        )
-    };
+            )
+        );
 
-    let make_write_instruction = |offset: u32, bytes: Vec<u8>| -> Instruction {
-        Instruction::new(
-            config.evm_loader,
-            &LoaderInstruction::Write {offset, bytes},
-            vec![AccountMeta::new(program_code, false),
-                 AccountMeta::new(creator.pubkey(), true)]
-        )
-    };
-
-    let make_finalize_instruction = || -> Instruction {
-        Instruction::new(
-            config.evm_loader,
-            &LoaderInstruction::Finalize,
-            vec![AccountMeta::new(program_id, false),
-                 AccountMeta::new(program_code, false),
-                 AccountMeta::new(caller, false),
-                 AccountMeta::new(creator.pubkey(), true),
-                 AccountMeta::new(clock::id(), false),
-                 AccountMeta::new(rent::id(), false),
-                 AccountMeta::new(config.evm_loader, false),
-                ]
-        )
+        send_transaction(config, &instructions)?;
     };
 
+    // Create transaction prepared for execution from account
+    let msg = {
+        let rlp_data = {
+            let tx = UnsignedTransaction {
+                to: None,
+                nonce: trx_count,
+                gas_limit: 1.into(),
+                gas_price: 1.into(),
+                value: 0.into(),
+                data: program_data,
+                chain_id: 111.into(),
```

Comment:
In the future, we will need to have the ability to set the `chain_id` when calling the utility. Please, create an issue for this and add a comment "Will fixed in #issue" to this line.